src/java.base/windows/classes/sun/nio/fs/WindowsPath.java

```diff
@@ -291,6 +307,13 @@ static String addPrefixIfNeeded(String path) {
         return path;
     }
 
+    static String addPrefixIfAbsent(String path) {
+        if (!path.startsWith("\\\\")) {
+            path = "\\\\?\\" + path;
+        }
```

Comment:
I think the approach looks okay but I wonder if we need to handle the UNC case here.

Reply:
Addressed in commit 01.

Reply:
It would be a lot simpler to just add "addPrefix" that unconditionally adds the prefix. There should be no scenario where double translation happens, if it does then there's a bug and I'm concerned that addPrefixIfAbsent will mask it.

Reply:
Updated in commit 02.